app/presenters/user_presenter.rb

```diff
@@ -156,6 +156,21 @@ def previous_names(template)
     user.user_name_change_requests.map { |req| template.link_to req.original_name, req }.join(", ").html_safe
   end
 
+  def favorite_tags_with_types
+    tag_names = user&.favorite_tags.to_s.split
+    tag_names = TagAlias.to_aliased(tag_names)
+    Tag.where(name: tag_names).map {|x| [x.name, x.post_count, x.category]}
+  end
+
+  def recent_tags_with_types
+    versions = PostArchive.where(updater_id: user.id).where("updated_at > ?", 1.hour.ago).order(id: :desc).limit(150)
```

Comment:
If `post_versions` is a big table, I would recommend a compound btree index on `(updater_id, updated_at)`. This doesn't seem to be present.

Reply:
I'm still skeptical of how this whole thing works, so I'm not sure if I want to, or need to rip it out, or if I should just add an index yet. There is elasticsearch indexing on this tables contents, but it isn't being used here(yet?)

This was just code I copied over because it was in the wrong place in another class, and also gets used there as well. I am going to see if I can consolidate the two later.